stream.go

```diff
@@ -202,7 +205,7 @@ func (r *Client) streamPrediction(ctx context.Context, prediction *Prediction, l
 					b := buf.Bytes()
 					buf.Reset()
 
-					event := SSEEvent{}
+					event := SSEEvent{Type: SSETypeDefault}
 					if err := event.decode(b); err != nil {
 						errChan <- err
```

Comment:
```suggestion
						errChan <- err
						continue
```
Wait, the problem isn't that we don't set the default correctly, it's that we try to use the event value even if the `decode` method failed. That's why I was seeing zero-valued event structs (the whole struct was zero-valued, not just the event type).

Another action we could take is:
```suggestion
						errChan <- err
						return err
```
returning an error from within an errGroup will cause the context to be canceled, so the other goroutines will exit as well if we do that. I think if we fail to parse an event that's a sign of something quite bad. But I can also see the merit in making as small a change to existing behavior as possible.

Reply:
Ah, I see what you mean. Thanks for explaining that.

Looking at `decode`, the only time it returns an error is when it produces invalid UTF-8 data. And I think you're right that if that's happening, that's a bad sign.

Since the channels are managed by the `stream` method, closing them here is the only way to stop the API client consumer from spewing garbage. So I think `return` is the right call, even if it's a bigger change.